src/Core/Impl/Text/Position.cs

```diff
@@ -39,7 +39,20 @@ public struct Position {
 
         public static bool operator >(Position p1, Position p2) => p1.line > p2.line || p1.line == p2.line && p1.character > p2.character;
         public static bool operator <(Position p1, Position p2) => p1.line < p2.line || p1.line == p2.line && p1.character < p2.character;
+        public static bool operator ==(Position p1, Position p2) => p1.Equals(p2);
+        public static bool operator !=(Position p1, Position p2) => !p1.Equals(p2);
 
+        public bool Equals(Position other) => line == other.line && character == other.character;
+
+        public override bool Equals(object obj) {
```

Comment:
Usually `?` for short method like => obj is Position other ? Equals(other) : false;